src/screens/swap/swap-form/swap-form.tsx

```diff
@@ -167,18 +170,41 @@ export const SwapForm: FC<SwapFormProps> = ({ inputToken }) => {
     }
   }, [searchValue, assetsList]);
 
+  const minimumReceivedAmount = useMemo(() => {
+    if (bestTradeWithSlippageTolerance.length > 0) {
+      const lastTradeOperation = bestTradeWithSlippageTolerance[bestTradeWithSlippageTolerance.length - 1];
+
+      return mutezToTz(lastTradeOperation.bTokenAmount, outputAssets.asset.decimals);
+    }
+
+    return new BigNumber(0);
+  }, [bestTradeWithSlippageTolerance, outputAssets.asset.decimals]);
+
+  useEffect(() => {
+    if (isDefined(bestTrade)) {
+      const bestTradeOutput = getTradeOutputAmount(bestTrade);
+
+      const outputTzAmount = bestTradeOutput
+        ? mutezToTz(bestTradeOutput, outputAssets.asset.decimals)
+        : new BigNumber(0);
+
+      const feeAmount = minimumReceivedAmount.minus(minimumReceivedAmount.multipliedBy(ROUTING_FEE_RATIO));
+      const finalAmount = outputTzAmount.minus(feeAmount);
+
+      setFieldValue('outputAssets.amount', tzToMutez(finalAmount, outputAssets.asset.decimals), false);
+    }
+  }, [bestTrade, minimumReceivedAmount, outputAssets.asset.decimals, outputAssetSlug, setFieldValue]);
+
```

Comment:
the same comments that you  have on extension 

https://github.com/madfish-solutions/templewallet-extension/pull/755